File: src/main/java/com/github/svegon/capi/event/render/tooltip/ChestTooltipCallback.java
package com.github.svegon.capi.event.render.tooltip;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.block.ChestBlock;
import net.minecraft.client.item.TooltipContext;
import net.minecraft.item.ItemStack;
import net.minecraft.text.Text;
import net.minecraft.world.BlockView;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@FunctionalInterface
public interface ChestTooltipCallback {
    Event<ChestTooltipCallback> EVENT = EventFactory.createArrayBacked(ChestTooltipCallback.class,
            (block,stack, world, tooltip, options) -> {}, listeners -> (block,stack, world, tooltip, context) -> {
        for (ChestTooltipCallback listener : listeners) {
            listener.appendChestTooltip(block, stack, world, tooltip, context);
        }
    });

    void appendChestTooltip(ChestBlock block, ItemStack stack, @Nullable BlockView world, List<Text> tooltip,
                            TooltipContext options);
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/render/tooltip/ChestTooltipCallback.kt
package io.github.svegon.capi.event.render.tooltip

import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.client.item.TooltipContext
import net.minecraft.text.Text
import net.minecraft.world.BlockView

fun interface ChestTooltipCallback {
    fun appendChestTooltip(
        block: ChestBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?,
        options: TooltipContext?
    )

    companion object {
        @JvmField
        val EVENT: Event<ChestTooltipCallback> = EventFactory.createArrayBacked<ChestTooltipCallback>(
            ChestTooltipCallback::class.java,
            ChestTooltipCallback { block: ChestBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, options: TooltipContext? -> }) { listeners: Array<ChestTooltipCallback> ->
            ChestTooltipCallback { block: ChestBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, context: TooltipContext? ->
                for (listener in listeners) {
                    listener.appendChestTooltip(block, stack, world, tooltip, context)
                }
            }
        }
    }
}
